### PR TITLE
Add config flow for Derivative

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adjusts Derivative documentation for the added configuration flow.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/68268
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/derivative.markdown [hunk 1]
```diff
@@ -12,16 +12,35 @@ ha_qa_scale: internal
...
 ha_domain: derivative
+ha_config_flow: true
 ha_platforms:
   - sensor
 ---
 
-The `derivative` platform creates a sensor that estimates the derivative of the values provided by a source sensor.
-Derivative sensors are updated upon changes of the **source**.
+The derivative integration creates a sensor that estimates the derivative of the
+values provided by another source sensor. Derivative sensors are updated upon changes of the **source sensor** used.
```
Comment (emontnemery) [inline source/_integrations/derivative.markdown]:
```suggestion
values provided by another sensor (the **source sensor**). Derivative sensors are updated upon changes of the **source sensor**.
```

File: source/_integrations/derivative.markdown [hunk 2]
```diff
@@ -12,16 +12,35 @@ ha_qa_scale: internal
...
+{% include integrations/config_flow.md %}
+{% configuration_basic %}
+Name:
+  description: The name the sensor should have. You can change it again later.
+Input sensor:
+  description: The entity providing numeric readings to create the derivative of.
+Precision:
+  description: Round the calculated integration value to at most N decimal places.
+Time window:
+  description: The time window in which to calculate the derivative. Derivatives in this window will be averaged with a Simple Moving Average algorithm weighted by time. This is for instance useful for a sensor that outputs discrete values, or to filter out short duration noise. By default the derivative is calculated between two consecutive updates without any smoothing.
```
Comment (emontnemery) [inline source/_integrations/derivative.markdown]:
```suggestion
  description: The time window in which to calculate the derivative. Derivatives in this window will be averaged with a simple moving average algorithm (SMA) weighted by time. This is for instance useful for a sensor that outputs discrete values, or to filter out short duration noise. By default the derivative is calculated between two consecutive updates without any smoothing.
```

File: source/_integrations/derivative.markdown [hunk 3]
```diff
@@ -12,16 +12,35 @@ ha_qa_scale: internal
...
+Metric Prefix:
+  description: Metric unit to prefix the derivative result ([Wikipedia](https://en.wikipedia.org/wiki/Unit_prefix)).
+Time unoit:
```
Comment (emontnemery) [inline source/_integrations/derivative.markdown]:
```suggestion
Time unit:
```

File: source/_integrations/derivative.markdown [hunk 4]
```diff
@@ -12,16 +12,35 @@ ha_qa_scale: internal
...
-Derivative sensors are updated upon changes of the **source**.
+The derivative integration creates a sensor that estimates the derivative of the
+values provided by another source sensor. Derivative sensors are updated upon changes of the **source sensor** used.
 
-## Configuration
+{% include integrations/config_flow.md %}
+{% configuration_basic %}
+Name:
+  description: The name the sensor should have. You can change it again later.
+Input sensor:
```
Comment (emontnemery) [inline source/_integrations/derivative.markdown:27]: Should we call it the "source sensor" instead?

File: source/_integrations/derivative.markdown [hunk 5]
```diff
@@ -12,16 +12,35 @@ ha_qa_scale: internal
...
+  description: The entity providing numeric readings to create the derivative of.
+Precision:
+  description: Round the calculated integration value to at most N decimal places.
+Time window:
+  description: The time window in which to calculate the derivative. Derivatives in this window will be averaged with a Simple Moving Average algorithm weighted by time. This is for instance useful for a sensor that outputs discrete values, or to filter out short duration noise. By default the derivative is calculated between two consecutive updates without any smoothing.
+Metric Prefix:
+  description: Metric unit to prefix the derivative result ([Wikipedia](https://en.wikipedia.org/wiki/Unit_prefix)).
+Time unoit:
+  description: SI unit of time of the derivative. If this parameter is set, the unit of measurement will be set like x/y where x is the unit of the input sensor given and y is the value given here.
```
Comment (emontnemery) [inline source/_integrations/derivative.markdown]:
```suggestion
  description: SI unit of time of the derivative. If this parameter is set, the unit of measurement will be set to **x/y** where **x** is the unit of the source sensor and **y** is the value of this parameter.
```

File: source/_integrations/derivative.markdown (hunk 4 above)
Comment (frenck) [reply source/_integrations/derivative.markdown:27]: 🤷  I think of it as an input (which matched terminology we use with Blueprints I guess). But, either way, is fine IMHO. Currently, it matches the Core PR.

**Review by emontnemery — APPROVED** [review]

**Pull request merged**